pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -206,27 +235,25 @@ func (r *reconcilerBase) createOrPatchDeployment(ctx context.Context, declared *
 		}
 		r.isAutopilotCluster = &isAutopilot
 	}
-
-	adjusted, err := adjustContainerResources(*r.isAutopilotCluster, declared, current)
+	dep, err := isDeploymentSame(*r.isAutopilotCluster, declared, unObjCurrent, reconcilerManagerAllowList, r.scheme)
 	if err != nil {
 		return controllerutil.OperationResultNone, err
 	}
-	if adjusted {
+	if dep.isDeploymentAdjusted {
```

Comment:
Given that the returned struct is always used for deployments, I don't think we need "deployment" in the field names. Also, usually in Go you forgo the "is-" prefix for boolean fields and only use them on methods/funcs that return a bool. So you could just make this `adjusted`.

That said, I don't really understand what "adjusted" means here without looking at the impl of `isDeploymentSame`. Does this mean it's the same or different?

Does `isDeploymentSame` really need to return a struct or can it just return the json patch data and a bool as two different return values?